module/modmanager/manager.go

```diff
@@ -1075,6 +1080,60 @@ func (m *module) checkReady(ctx context.Context, parentAddr string, logger loggi
 	}
 }
 
+// FirstRun is runs a module-specific setup script.
+func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+	logger := mgr.logger.AsZap().With("name", conf.Name)
+
+	// Evaluate the Module's FirstRun path. If there is an error we assume
+	// that the first run script does not exist and we debug log and exit quietly.
+	firstRunPath, markSuccess, err := conf.EvaluateFirstRunPath(packages.LocalPackagesDir(mgr.packagesDir))
+	if err != nil {
+		// TODO(RSDK-9067): some first run path evaluation errors should be promoted to WARN logs.
+		logger.Debug("no first run script detected, skipping setup phase", "error", err)
+		return nil
+	}
+
+	logger.Info("executing first run script")
+
+	// This value is normally set on a field on the [module] struct but it seems like we can safely get it on demand.
+	var dataDir string
+	if mgr.moduleDataParentDir != "" {
+		var err error
+		// TODO: why isn't conf.Name being sanitized like PackageConfig.SanitizedName?
+		dataDir, err = rutils.SafeJoinDir(mgr.moduleDataParentDir, conf.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	moduleEnvironment := getFullEnvironment(conf, dataDir, mgr.viamHomeDir)
+
+	// TODO(RSDK-9060): support a user-supplied timeout
+	cmdCtx, cancel := context.WithTimeout(ctx, defaultFirstRunTimeout)
+	defer cancel()
+
+	//nolint:gosec // Yes, we are deliberating executing arbitrary user code here.
+	cmd := exec.CommandContext(cmdCtx, firstRunPath)
+
+	cmd.Env = os.Environ()
+	for key, val := range moduleEnvironment {
+		cmd.Env = append(cmd.Env, key+"="+val)
+	}
+	cmdOut, err := cmd.CombinedOutput()
+
+	resultLogger := logger.With("path", firstRunPath, "output", string(cmdOut))
```

Comment:
likely a followup, but this output does not show on app.viam logs. An option would be to dump the output as part of a Infof or Errorf log

Reply:
oh i think this is because i did the AsZap logger conversion in order to use the `With` method. that’s annoying… oh well, yeah i’ll follow up on this